src/tests/unit/platform/test_policy_ruleconfigs.py

```diff
@@ -280,3 +289,265 @@ def on_delete(url, body):
     assert rule_config.name == 'Carbon Black Threat Intel'
     rule_config.delete()
     assert delete_called
+
+
+def test_host_based_firewall_contents(cbcsdk_mock):
+    """Tests the contents of the host-based firewall rule configuration, along with the refresh."""
+    cbcsdk_mock.mock_request('GET', '/policyservice/v1/orgs/test/policies/65536/rule_configs/host_based_firewall',
+                             HBFW_GET_RESULT)
+    api = cbcsdk_mock.api
+    policy = Policy(api, 65536, copy.deepcopy(FULL_POLICY_1), False, True)
+    rule_config = policy.host_based_firewall_rule_config
+    assert not rule_config.enabled
+    assert rule_config.default_action == "ALLOW"
+    groups = rule_config.rule_groups
+    assert len(groups) == 1
+    assert groups[0].name == "Argon_firewall"
+    assert groups[0].description == "Whatever"
+    rules = groups[0].rules_
+    assert len(rules) == 1
+    assert rules[0].action == "ALLOW"
```

Comment:
Instead of verifying each property individually why not use `rule_config._info == HBFW_GET_RESULT["results"][0]` then step into the sub objects and do further step in comparisons like `groups._info == HBFW_GET_RESULT["results"][0]["parameters]["rulesets"]`

This also seems to be doing two tests at once with initial parsing but also a refresh which should use alternative values which would change before and after the refresh. I'd recommend these being different tests

Reply:
That's fair though in this case let's not check every property as it leads to bloated tests and is not increasing code coverage

Reply:
Agreed. I can certainly omit a few of those without sacrificing generality.  Same would apply to the post-refresh test, when it gets split out.

Reply:
I split the test and pared down the number of properties that were checked (by about half, I think, in the case of the "contents" test). I was even able to simplify the "contents" test a bit as a result. This will be in the next commit.